sw/airborne/modules/osd_max7456_new/max7456.c

```diff
@@ -0,0 +1,1364 @@
+/*
+ * Copyright (C) 2013 Chris
+ *
+ * This file is part of paparazzi.
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+/*  MAX 7456 OSD DRIVER
+*   Date 21 Aug 2013
+*   VERSION 1.1
+*   The driver now checks the busy flag and it can justify the characters in left, right or center.
+*/ 
+#include "std.h"
+#include "stdio.h"
+
+#include <math.h>
+#include <inttypes.h>
+#include "inter_mcu.h"
+
+#include "mcu_periph/sys_time.h"
+#include "mcu_periph/gpio.h"
+#include "mcu_periph/spi.h"
+#include "led.h"
+#include "autopilot.h"
+
+#include "subsystems/navigation/common_nav.h"
+#include "firmwares/fixedwing/nav.h"
+#include "generated/flight_plan.h"
+#include "generated/airframe.h"
+#include "subsystems/datalink/datalink.h"
+#include "subsystems/electrical.h"
+
+//#include "messages.h"
+#include "subsystems/datalink/downlink.h"
+#include "state.h"
+
+// Peripherials
+#include "max7456.h"
+
+#if !defined(SITL)
+
+#if defined(STM32) || defined(BOARD_KROOZ)
```

Comment:
all these "automatic" defines should come from the module configuration, eventually with a default to the conf of the Matek board if it is made for this board